### PR TITLE
fix!: remove `puppeteer.devices` in favor of `KnownDevices`

### PR DESCRIPTION
**Pull request by jrandolf-google**
This PR removes the deprecated `puppeteer.devices` in favor of a new exported object `KnownDevices`. `devices` can also be exported, but has been deprecated.

### CODE REVIEW

File: packages/puppeteer-core/src/api/Page.ts [hunk 1]
```diff
@@ -1679,21 +1685,16 @@ export class Page extends EventEmitter {
    *   await browser.close();
    * })();
    * ```
-   *
-   * @remarks List of all available devices is available in the source code:
```
Comment (OrKoN) [inline packages/puppeteer-core/src/api/Page.ts:1683]: let's still link to the full list of devices

Comment (OrKoN) [reply packages/puppeteer-core/src/api/Page.ts:1683]: ah is the link in the beginning now?

File: packages/puppeteer-core/src/common/Device.ts [hunk 2]
```diff
@@ -14,23 +14,24 @@
...
-    isLandscape: boolean;
-  };
+  viewport: Viewport;
+}
+
+/**
+ * @public
+ */
+export interface KnownDevice extends Device {
+  name: string;
```
Comment (OrKoN) [inline packages/puppeteer-core/src/common/Device.ts]: Let's keep the name to be on the Device interface? Then there would be no need for a breaking change and a KnownDevice type.

File: packages/puppeteer-core/src/api/Page.ts (hunk 1 above)
Comment (jrandolf) [reply packages/puppeteer-core/src/api/Page.ts:1683]: Yes, the link is in the beginning now :)

File: packages/puppeteer-core/src/common/Device.ts (hunk 2 above)
Comment (jrandolf) [reply packages/puppeteer-core/src/common/Device.ts]: Done.

**Review by OrKoN — APPROVED** [review]

**Comment by Kikobeats** [comment]
is this change promoted for a reason?

I'm not sure why `KnownDevices` and not `knownDevices` 🤔

**Pull request merged**